pkg/logging/kafka/create.go

```diff
@@ -124,6 +136,35 @@ func (c *CreateCommand) createInput() (*fastly.CreateKafkaInput, error) {
 		input.Placement = fastly.String(c.Placement.Value)
 	}
 
+	if c.ParseLogKeyvals.Valid {
+		input.ParseLogKeyvals = fastly.CBool(c.ParseLogKeyvals.Value)
+	}
+
+	if c.RequestMaxBytes.Valid {
+		input.RequestMaxBytes = fastly.Uint(c.RequestMaxBytes.Value)
+	}
+
+	if c.UseSASL.Valid && c.UseSASL.Value {
+		if c.AuthMethod.Value == "" || c.User.Value == "" || c.Password.Value == "" {
+			return nil, fmt.Errorf("the --auth-method, --username,  and --password flags must be present when using the --use-sasl flag")
+
+		}
+
+		input.AuthMethod = fastly.String(c.AuthMethod.Value)
+
+		if c.User.Valid {
+			input.User = fastly.String(c.User.Value)
+		}
+
+		if c.Password.Valid {
+			input.Password = fastly.String(c.Password.Value)
+		}
```

Comment:
Can be outside of the surrounding `if c.UseSASL.Valid && c.UseSASL.Value {` conditional.

Reply:
Done